src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java

```diff
@@ -48,4 +50,23 @@ public SnapshotBuilderCountResponse getCountResponse(
         .sql("")
         .result(new SnapshotBuilderCountResponseResult().total(getRollupCount(id, cohorts)));
   }
+
+  public EnumerateSnapshotAccessRequest enumerateByDatasetId(UUID id) {
+    return convertToEnumerateModel(snapshotRequestDao.enumerateByDatasetId(id));
+  }
+
+  public EnumerateSnapshotAccessRequest convertToEnumerateModel(
```

Comment:
If this is only made public so that you can test it, I'd advise annotating it with `@VisibleForTesting` and removing the `public` access modifier.